src/main/java/us/myles/ViaVersion/protocols/protocol1_9to1_8/ItemRewriter.java

```diff
@@ -245,10 +243,72 @@ public static void toClient(Item item) {
         }
     }
 
+    public static String potionNameFromDamage(short damage) {
+        String cached = POTION_ID_TO_NAME.get((int) damage);
+        if(cached != null) {
+            return cached;
+        }
+        if(damage == 0) {
+            return "water";
+        }
+
+        int effect = damage & 0xF;
+        int name = damage & 0x3F;
+        boolean enhanced = (damage & 0x20) > 0;
+        boolean extended = (damage & 0x40) > 0;
+
+        boolean canEnhance = true;
+        boolean canExtend = true;
+
+        String id;
+        switch (effect) {
```

Comment:
I'd prefer this formatted to the java standards,

http://www.oracle.com/technetwork/java/codeconventions-150003.pdf